app/configuration.py

```diff
@@ -179,17 +187,24 @@ def _validate_config(config: dict):
                     item in config[req_key] and config[req_key][item]
                 ), f"No {item!r} provided in {req_key!r} section of configuration"
 
-        def _is_valid_command_topic(topic):
+        def _is_valid_mqtt_topic(topic):
             # Command topic must be str, must not have more than 7 levels
-            return (
-                isinstance(config["mqtt_settings"]["command_topic"], str)
-                and config["mqtt_settings"]["command_topic"].count("/") <= 7  # noqa
-            )
+            return isinstance(topic, str) and topic.count("/") <= 7
```

Comment:
we probably also want to ensure the topic name doesn't contain `+` and `#` when it comes to _publishing_ messages.
For subscription, those symbols are valid, as those represent wildcard. But not for publishing. It should be "fully qualified"